rc/scdoc.kak

```diff
@@ -0,0 +1,54 @@
+# https://git.sr.ht/~sircmpwn/scdoc
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+# Detection
+# ‾‾‾‾‾‾‾‾‾
+
+hook global BufCreate .*[.]scd %{
+    set-option buffer filetype scdoc
+}
+
+# Initialization
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+hook global WinSetOption filetype=scdoc %{
+    require-module scdoc
+    hook -once -always window WinSetOption filetype=.* %{ remove-hooks window scdoc-.+ }
+}
+
+hook -group scdoc-highlight global WinSetOption filetype=scdoc %{
+    add-highlighter window/scdoc ref scdoc
+    hook -once -always window WinSetOption filetype=.* %{ remove-highlighter window/scdoc}
+}
+
+
+provide-module scdoc %{
+
+# Highlighters
+# ‾‾‾‾‾‾‾‾‾‾‾‾
+
+add-highlighter shared/scdoc regions
+add-highlighter shared/scdoc/inline default-region regions
+add-highlighter shared/scdoc/inline/text default-region group
+
+add-highlighter shared/scdoc/inline/text/ regex \
+	'(?<!# )((?<!\\)|\\\\)\*(\n?[^\n])*?((?<!\\)|\\\\)\*' 0:bold
+add-highlighter shared/scdoc/inline/text/ regex \
+	'(?<!# )((?<!\\)|\\\\)\b_(\n?[^\n])*?((?<!\\)|\\\\)_\b' 0:italic
+# Let’s break this down:
+# (?<!# ) Tretead literally in headings
+# ((?<!\\)|\\\\) Treated literally after \ but not after \\
+# (\n?[^\n]) Line break is allowed but not a paragraph break
```

Comment:
Have you tried using the regex I set in [fountain.kak](https://github.com/kakoune-editor/kakoune-extra-filetypes/blob/master/rc/fountain.kak#L33-L36)? They are stolen from upstream, and seem to work quite well.

Your regex are slightly different and still have issues (c.f. the TODO below), maybe you could give the regex linked above a try?

Reply:
The TODO is an extreme edge case and the fountain.kak regex does not fix it. scdoc’s syntax is also a bit simpler: it allows spaces around `*` and `_` for example.